aspnetcore/mvc/views/tag-helpers/th-components/samples/RazorPagesSample/Pages/Index.cshtml.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace RazorPagesSample.Pages
+{
+    #region snippet_IndexModelClass
+    using System;
```

Comment:
I think I understand that you're doing this to show the namespaces above the code example so the reader can see what they need to add without the namespace getting in the way. My concern tho is that the sample is now showing a non-standard class file structure.

I wonder (hope not) if any devs, particularly new devs, will see this pattern and *start doing it in real life*. I'm almost inclined to say that each time this is done that a code comment should appear *above the `region` line* (so it doesn't show) that explains why this is being done.